File: sy43/app/src/main/java/com/example/sy43/database/AppDatabase.java
package com.example.sy43.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Category.class,SubCategory.class, ExtraDebts.class,MonthlyRevenue.class,Expenses.class,Projects.class,Bills.class,Income.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    public abstract CategoryDao categoryDao();
    public abstract SubCategoryDao subCategoryDao();
    public abstract ExtraDebtsDao extraDebtDao();
    public abstract MonthlyRevenueDao monthlyRevenueDao();
    public abstract ExpensesDao expensesDao();
    public abstract ProjectsDao projectDao();
    public abstract BillsDao billsDao();
    public abstract IncomeDao incomeDao();

    private static AppDatabase instance;
//    private static final int NUMBER_OF_THREADS = 4;
//    static final ExecutorService databaseWriteExecutor = Executors.newFixedThreadPool(NUMBER_OF_THREADS);

    public static synchronized AppDatabase getInstance(Context context){
        if (instance == null){
            instance = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "Padnom").allowMainThreadQueries().build();
        }
        return instance;
    }


    //AppDatabase db = Room.databaseBuilder(getApplicationContext(),AppDatabase.class, "Padnom").build();

    //CategoryDao categoryDao = db.categoryDao();
}

File: src/main/java/com/example/testbare/database/AppDatabase.kt
package com.example.testbare.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.testbare.database.dao.CategorieDao
import com.example.testbare.database.dao.DepenseDao
import com.example.testbare.database.entities.Categorie
import com.example.testbare.database.entities.Depense

@Database(
    entities = [
        Depense::class,
        Categorie::class
    ],
    version = 1
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun CategorieDao() : CategorieDao
    abstract fun DepenseDao() : DepenseDao

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getDatabase(context: Context): AppDatabase {
            context.deleteDatabase("app_database4")
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context,
                    AppDatabase::class.java,
                    "app_database")
                    .build()
                INSTANCE = instance

                instance
            }
        }
    }
}